src/client/main.ts

```diff
@@ -0,0 +1,280 @@
+    source: string;
+    data?: unknown;
+    version: string;
+}
+
+export type Listener = ContextListener|IntentListener;
+
+export interface ContextListener {
+    handler: (context: Context) => void;
+    /**
+     * Unsubscribe the listener object.
+     */
+    unsubscribe: () => void;
+}
+
+export interface IntentListener {
+    intent: string;
+    handler: (context: Context) => void;
+    /**
+     * Unsubscribe the listener object.
+     */
+    unsubscribe: () => void;
+}
+
+/**
+ * A Desktop Agent is a desktop component (or aggregate of components) that serves as a
+ * launcher and message router (broker) for applications in its domain.
+ *
+ * A Desktop Agent can be connected to one or more App Directories and will use directories for application
+ * identity and discovery. Typically, a Desktop Agent will contain the proprietary logic of
+ * a given platform, handling functionality like explicit application interop workflows where
+ * security, consistency, and implementation requirements are proprietary.
+ */
+
+
+/**
+ * Launches/links to an app by name.
+ *
+ * If a Context object is passed in, this object will be provided to the opened application via a contextListener.
+ * The Context argument is functionally equivalent to opening the target app with no context and broadcasting the context directly to it.
+ *
+ * If opening errors, it returns an `Error` with a string from the `OpenError` export enumeration.
+ *
+ *  ```javascript
+ *     //no context
+ *     agent.open('myApp');
+ *     //with context
+ *     agent.open('myApp', context);
+ * ```
+ */
+export async function open(name: string, context?: Context): Promise<void> {
+    return tryServiceDispatch(APITopic.OPEN, {name, context});
+}
+
+/**
+ * Find out more information about a particular intent by passing its name, and optionally its context.
+ *
+ * findIntent is effectively granting programmatic access to the Desktop Agent's resolver.
+ * A promise resolving to the intent, its metadata and metadata about the apps that registered it is returned.
+ * This can be used to raise the intent against a specific app.
+ *
+ * If the resolution fails, the promise will return an `Error` with a string from the `ResolveError` export enumeration.
+ *
+ * ```javascript
+ * // I know 'StartChat' exists as a concept, and want to know more about it ...
+ * const appIntent = await agent.findIntent("StartChat");
+ *
+ * // returns a single AppIntent:
+ * // {
+ * //     intent: { name: "StartChat", displayName: "Chat" },
+ * //     apps: [{ name: "Skype" }, { name: "Symphony" }, { name: "Slack" }]
+ * // }
+ *
+ * // raise the intent against a particular app
+ * await agent.raiseIntent(appIntent.intent.name, context, appIntent.apps[0].name);
+ * ```
+ */
+export async function findIntent(intent: string, context?: Context): Promise<AppIntent> {
+    return tryServiceDispatch(APITopic.FIND_INTENT, {intent, context});
+}
+
+/**
+ * Find all the avalable intents for a particular context.
```

Comment:
available*

Reply:
Those comments were all copy/pasted from the FDC3 spec docs :P 

I'll update, and maybe should tell them too.